packages/app/src/cli/models/app/loader.ts

```diff
@@ -269,17 +269,35 @@ class AppLoader {
     const webTomlPaths = await glob(webConfigGlobs)
 
     const webs = await Promise.all(webTomlPaths.map((path) => this.loadWeb(path)))
+    this.validateWebs(webs)
 
     const webTomlsInStandardLocation = await glob(joinPath(this.appDirectory, `web/**/${configurationFileNames.web}`))
     const usedCustomLayout = webDirectories !== undefined || webTomlsInStandardLocation.length !== webTomlPaths.length
 
     return {webs, usedCustomLayout}
   }
 
+  validateWebs(webs: Web[]): void {
+    ;[WebType.Backend, WebType.Frontend].forEach((webType) => {
+      const websOfType = webs.filter((web) => web.configuration.roles.includes(webType))
+      if (websOfType.length > 1) {
+        this.abortOrReport(
+          outputContent`You can only have one web with the ${outputToken.yellow(webType)} role in your app`,
+          undefined,
+          websOfType[1]!.directory,
+        )
+      }
+    })
+  }
+
   async loadWeb(WebConfigurationFile: string): Promise<Web> {
+    const config = await this.parseConfigurationFile(WebConfigurationSchema, WebConfigurationFile)
+    const roles = new Set('roles' in config ? config.roles : [])
```

Comment:
would this work?
```suggestion
    const roles = new Set(config.roles ?? [])
```

Reply:
@isaacroldan this is a typescript thing, because it's a union type you can never just say `config.roles` but instead you have to establish that we're in this situation where `roles` exists.